declaring empty enums results in a C error

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relvant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** V 0.2.1 954580f.095857d
**OS:** Linux Mint 20

**What did you do?**
I have declared an empty enum:
```v
Welcome to the V REPL (for help with V itself, type `exit`, then run `v help`).
V 0.2.1 760de7c
Use Ctrl-C or `exit` to exit, or `help` to see other available commands
>>> enum Type {}
```

**What did you expect to see?**
An error from V saying that the enum cannot be empty.
 
**What did you see instead?**
A C error:
```
/tmp/v/.noprefix.vrepl_temp.11313248834576623734.tmp.c:759:1: error: empty enum is invalid
  759 | } main__Type;
      | ^
```

same
```v
enum Enm {}

fn main() {
}
```
```
/tmp/v/main.9828732497014025660.tmp.c:570:1: error: empty enum is invalid
  570 | } main__Enm;
      | ^
```

archlinux
V 0.2.1 43adbf4

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -2194,6 +2194,9 @@ pub fn (mut c Checker) const_decl(mut node ast.ConstDecl) {
 pub fn (mut c Checker) enum_decl(decl ast.EnumDecl) {
 	c.check_valid_pascal_case(decl.name, 'enum name', decl.pos)
 	mut seen := []i64{}
+	if decl.fields.len == 0 {
+		c.error('enum cannot be empty', decl.pos)
+	}
 	for i, field in decl.fields {
 		if !c.pref.experimental && util.contains_capital(field.name) {
 			// TODO C2V uses hundreds of enums with capitals, remove -experimental check once it's handled

--- a/vlib/v/checker/tests/enum_empty.out
+++ b/vlib/v/checker/tests/enum_empty.out
@@ -1,0 +1,3 @@
+vlib/v/checker/tests/enum_empty.vv:1:1: error: enum cannot be empty
+    1 | enum Empty {}
+      | ~~~~~~~~~~

--- a/vlib/v/checker/tests/enum_empty.vv
+++ b/vlib/v/checker/tests/enum_empty.vv
@@ -1,0 +1,1 @@
+enum Empty {}

